visivo/query/insight_tokenizer.py

```diff
@@ -0,0 +1,569 @@
+        pre_query = self._generate_pre_query()
+        post_query = self._generate_post_query()
+
+        if self.is_dynamic_interactions:
+            pre_query, post_query = post_query, pre_query
+
+        # Determine source type
+        if isinstance(self.model, LocalMergeModel):
+            source_type = "duckdb"
+        elif self.source.type:
+            source_type = self.source.type
+        else:
+            source_type = None
+
+        if self.insight.props:
+            props = self.insight.props.model_dump()
+        else:
+            props = None
+
+        return TokenizedInsight(
+            name=self.insight.name,
+            source=self.source.name,
+            source_type=source_type,
+            description=self.insight.description,
+            pre_query=pre_query,
+            post_query=post_query,
+            select_items=self.select_items,
+            column_items=self.column_items,
+            selects=self.selects,
+            columns=self.columns,
+            props=props,
+            interactions=self._serialize_interactions(),
+            input_dependencies=list(self.input_dependencies),
+            requires_groupby=len(self.groupby_statements) > 0,
+            groupby_statements=list(self.groupby_statements) if self.groupby_statements else None,
+            split_column=self._get_split_column(),
+            sort_expressions=self._get_sort_expressions(),
+        )
+
+    def _analyze_insight(self):
+        """Master analysis method that coordinates all parsing"""
+        self._analyze_props()
+        self._analyze_columns()
+        self._analyze_interactions()
+        self._determine_groupby_requirements()
+
+    def _analyze_props(self):
+        """Extract SQL expressions from insight props using recursive traversal"""
+        self._extract_select_items(self.insight.props, ["props"])
+
+    def _analyze_columns(self):
+        """Extract SQL expressions from insight columns section"""
+        if self.insight.columns:
+            self._extract_select_items(self.insight.columns, ["columns"])
+
+    def _extract_select_items(self, obj: Any, path: List[str]):
+        """Recursively extract SQL expressions from nested objects"""
+        if obj is None:
+            return
+
+        if isinstance(obj, (InsightProps, Layout)):
+            # Get all fields including extra ones from the model dump
+            props_dict = obj.model_dump()
+            for key, value in props_dict.items():
+                if value is not None:
+                    self._extract_select_items(value, path + [key])
+
+        elif isinstance(obj, BaseModel):
+            for prop in obj.__class__.model_fields.keys():
+                prop_value = getattr(obj, prop, None)
+                if prop_value is not None:
+                    self._extract_select_items(prop_value, path + [prop])
+
+        elif isinstance(obj, InsightColumns):
```

Comment:
After we get the metrics items in, we will probably remove columns in favor of the dimentions higher up in the dag.